src/CMakeLists.txt

```diff
@@ -25,18 +25,14 @@ add_library(runtime
   config.cpp
   disasm.cpp
   dwarf_parser.cpp
-  fake_map.cpp
   format_string.cpp
-  imap.cpp
   log.cpp
-  map.cpp
   mapkey.cpp
   output.cpp
   probe_matcher.cpp
   procmon.cpp
   printf.cpp
   resolve_cgroupid.cpp
-  required_resources.cpp
```

Comment:
This means from core bpftrace files we only use required_resources.h definitions. Meaning if future pubic methods are exposed and called from core bpftrace, then you'd get a linker error then.

What was the linker error that motivated this change? Curious cuz I feel like it should work.

Reply:
Since no function from `required_resources.cpp` is used by non-aot code, the linker optimizes it away when linking `libruntime` and afterwards linking `libaot` fails with:
```
/usr/bin/ld: aot/libaot.a(aot.cpp.o): in function `bpftrace::aot::(anonymous namespace)::load_required_resources(bpftrace::BPFtrace&, unsigned char*, unsigned long)':
/home/vmalik/src/bpf/bpftrace/src/aot/aot.cpp:77: undefined reference to `bpftrace::RequiredResources::load_state(unsigned char const*, unsigned long)'
/usr/bin/ld: aot/libaot.a(aot.cpp.o): in function `bpftrace::aot::(anonymous namespace)::generate_btaot_section(bpftrace::RequiredResources const&, bpftrace::BpfBytecode const&)':
/home/vmalik/src/bpf/bpftrace/src/aot/aot.cpp:109: undefined reference to `bpftrace::RequiredResources::save_state(std::ostream&) const'
```
My original solution was to add a dummy constructor of `RequiredResrouces` to `required_resources.cpp` but @ajor had a good point that it doesn't make sense linking an object file from which no code is used. OTOH, if we ever added some code needed from `libruntime`, we'd have to revert all this.